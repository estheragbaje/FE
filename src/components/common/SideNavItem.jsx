import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import {
  smallSpace, mediumSpace2, black, blue,
} from '../../styles/variables';

const SideNavItem = ({ icon, text, path }) => (
  <StyledNavItem>
    <NavLink to={path} activeClassName="active-nav"><i className="material-icons">{icon}</i></NavLink>
    <NavLink to={path} activeClassName="active-nav"><span>{text}</span></NavLink>
  </StyledNavItem>
);

const StyledNavItem = styled.div`
    a {
        color: ${black};
        text-decoration: none;
    }

    display: flex;
    align-items: center;
    background: #F9FDFD;

    i {
        margin-right: ${smallSpace};
        vertical-align: middle;
    }

    .active-nav {
        color: ${blue}
    }

    margin-bottom: ${mediumSpace2};
`;

export default SideNavItem;
