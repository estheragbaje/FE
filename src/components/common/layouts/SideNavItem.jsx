import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const SideNavItem = ({ icon, text, path }) => (
  <StyledNavItem>
    <NavLink to={path} activeClassName="active-nav">
      <span className="label">
        <i className="material-icons">{icon}</i>
        &nbsp; &nbsp;
        <span>{text}</span>
      </span>
    </NavLink>
  </StyledNavItem>
);

SideNavItem.propTypes = {
  icon: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};

const StyledNavItem = styled.div`
  display: flex;
  align-items: center;

  span.label {
    display: flex;
    align-items: center;
  }

  a {
    display: block;
    width: 100%;
    padding: 1rem;
    padding-left: 1.4rem;
    border-bottom: 1px solid #cfedfe;
    text-decoration: none;
    color: #444;
    line-height: 1.4rem;
  }

  &:hover a {
    color: blue;
    background: #dbf2ff;
  }
`;

export default SideNavItem;
