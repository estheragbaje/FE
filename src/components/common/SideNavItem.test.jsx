import React from 'react';
import { shallow } from 'enzyme';

import SideNavItem from './SideNavItem';

describe('<SideNavItem />', () => {
  const props = {
    icon: jest.mock(),
    text: jest.mock(),
    path: jest.mock(),
  };

  it('renders the SideNavItem component correctly', () => {
    const wrapper = shallow(<SideNavItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
