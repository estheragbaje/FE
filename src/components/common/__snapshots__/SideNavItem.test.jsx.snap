// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavItem /> renders the SideNavItem component correctly 1`] = `ShallowWrapper {}`;
